pymeasure/instruments/mock.py

```diff
@@ -100,3 +106,52 @@ def set_output_voltage(self, value):
         self._output_voltage = value
 
     output_voltage = property(fget=get_output_voltage, fset=set_output_voltage)
+
+    def get_frame_width(self):
+        """ Image frame width in pixels."""
+        time.sleep(self._wait) 
+        return self._w
+
+    def set_frame_width(self, w):
+        time.sleep(self._wait)
+        self._w = w
+
+    frame_width = property(fget=get_frame_width, fset=set_frame_width)
+
+    def get_frame_height(self):
+        """ Image frame height in pixels."""
+        time.sleep(self._wait) 
+        return self._h
+
+    def set_frame_height(self, h):
+        time.sleep(self._wait)
+        self._h = h
+
+    frame_height = property(fget=get_frame_height, fset=set_frame_height)
+
+    def get_frame_format(self):
+        """ Format for image data returned from the get_frame() method. Allowed values are:
+                mono_8: single channel 8-bit image.
+                mono_16: single channel 16-bit image.
+        """
+        time.sleep(self._wait) 
+        return self._frame_format 
+
+    def set_frame_format(self, form):
+        if not form in ["mono_8", "mono_16"]:
```

Comment:
you could use our `strict_discrete_set` [validator](https://pymeasure.readthedocs.io/en/latest/api/instruments/validators.html) for this.

Reply:
Yep. Using this now